packages/react-dev-utils/webpackHotDevClient.js

```diff
@@ -55,39 +52,74 @@ function showErrorOverlay(message) {
   iframe.style.height = '100vh';
   iframe.style.border = 'none';
   iframe.style.zIndex = 9999999999;
-  document.body.appendChild(iframe);
-
-  // Inside, make a div.
-  var overlayID = 'react-dev-utils-webpack-hot-dev-client-overlay-div';
-  var overlay =
-    iframe.contentDocument.getElementById(overlayID) ||
-    iframe.contentDocument.createElement('div');
-  overlay.id = overlayID;
-  overlay.style.position = 'fixed';
-  overlay.style.left = 0;
-  overlay.style.top = 0;
-  overlay.style.right = 0;
-  overlay.style.bottom = 0;
-  overlay.style.width = '100vw';
-  overlay.style.height = '100vh';
-  overlay.style.backgroundColor = 'black';
-  overlay.style.color = '#E8E8E8';
-  overlay.style.fontFamily = 'Menlo, Consolas, monospace';
-  overlay.style.fontSize = 'large';
-  overlay.style.padding = '2rem';
-  overlay.style.lineHeight = '1.2';
-  overlay.style.whiteSpace = 'pre-wrap';
-  overlay.style.overflow = 'auto';
-
-  // Make it look similar to our terminal.
-  overlay.innerHTML =
-    '<span style="color: #' +
-    colors.red +
-    '">Failed to compile.</span><br><br>' +
-    ansiHTML(entities.encode(message));
-
-  // Render!
-  iframe.contentDocument.body.appendChild(overlay);
+  iframe.onload = onIframeLoad;
+  return iframe;
+}
+
+function addOverlayDivTo(iframe) {
+  var div =  iframe.contentDocument.createElement('div');
+  div.id = 'react-dev-utils-webpack-hot-dev-client-overlay-div';
+  div.style.position = 'fixed';
+  div.style.left = 0;
+  div.style.top = 0;
+  div.style.right = 0;
+  div.style.bottom = 0;
+  div.style.width = '100vw';
+  div.style.height = '100vh';
+  div.style.backgroundColor = 'black';
+  div.style.color = '#E8E8E8';
+  div.style.fontFamily = 'Menlo, Consolas, monospace';
+  div.style.fontSize = 'large';
+  div.style.padding = '2rem';
+  div.style.lineHeight = '1.2';
+  div.style.whiteSpace = 'pre-wrap';
+  div.style.overflow = 'auto';
+  iframe.contentDocument.body.appendChild(div);
+  return div;
+}
+
+var overlayIframe = null;
+var overlayDiv = null;
+var lastOnOverlayDivReady = null;
+
+function ensureOverlayDivExists(onOverlayDivReady) {
+  if (overlayDiv) {
+    // Everything is ready, call the callback right away.
+    onOverlayDivReady(overlayDiv);
+    return;
+  }
+
+  // Creating an iframe may be asynchronous so we'll schedule the callback.
+  // In case of multiple calls, last callback wins.
+  lastOnOverlayDivReady = onOverlayDivReady;
+
+  if (overlayIframe) {
+    // We're already creating it.
+    return;
+  }
+
+  // Create iframe and, when it is ready, a div inside it.
+  overlayIframe = createOverlayIframe(function onIframeLoad() {
+    overlayDiv = addOverlayDivTo(overlayIframe);
+    // Now we can talk!
+    lastOnOverlayDivReady(overlayDiv);
+  });
+
+  // Zalgo alert: onIframeLoad() will be called either synchronouly
```

Comment:
I ran into this issue when trying to setup dom measurement for css-layout browser testing. This was super annoying :)